
syntax = "proto3";

import "experimental/beacon_sim/experiment_config.proto";
import "common/time/robot_time.proto";
import "common/math/matrix.proto";

package robot.experimental.beacon_sim.proto;

message Plan {
  repeated int32 nodes = 1;
  double log_prob_mass = 2;
  double expected_det = 3;
}

message PlannerResult {
  int32 trial_id = 1;
  int32 planner_id = 2;
  robot.time.proto.RobotTimestampDuration elapsed_time = 3;
  optional Plan plan = 4;
}

message StartGoal {
  robot.math.proto.Matrix start = 1;
  robot.math.proto.Matrix goal = 2;
}

message ExperimentResult {
  robot.experimental.beacon_sim.proto.ExperimentConfig experiment_config = 1;
  repeated StartGoal start_goal = 2;
  repeated string planner_names = 3;
  repeated PlannerResult results = 4;
}

